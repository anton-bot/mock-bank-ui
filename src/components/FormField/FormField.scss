.FormField {
  .label {
    font-weight: bold;
    margin-bottom: 0.25em;
  }

  .control {
    input,
    select {
      padding: 0.5em;
      border-radius: 0.25em;
      border: 1px solid lightgray;
      width: 100%;
      box-sizing: border-box;
    }
  }

  .error {
    color: crimson;
    font-size: 0.85em;
  }
}
