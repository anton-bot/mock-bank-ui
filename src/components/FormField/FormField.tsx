import './FormField.scss';

type Props = {
  label: string;
  error: string | undefined;
  touched: boolean;
};

export const FormField: React.FC<Props> = (props) => {
  return (
    <div className="FormField">
      <div className="label">{props.label}</div>
      <div className="control">{props.children}</div>
      {props.touched && props.error && <div className="error">{props.error}</div>}
    </div>
  );
};
