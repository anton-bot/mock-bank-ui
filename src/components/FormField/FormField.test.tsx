import { render } from '@testing-library/react';
import { FormField } from './FormField';
import renderer from 'react-test-renderer';

describe('Form Field', () => {
  it('must show an error when field was edited and input is invalid', () => {
    const { getByText } = render(
      <FormField label="Account Number" error="Invalid account number" touched={true}>
        <input />
      </FormField>,
    );

    expect(getByText(/Invalid account number/)).toBeInTheDocument();
  });

  it('must not show an error when field has an invalid input, but it was not edited', () => {
    const { queryByText } = render(
      <FormField label="Account Number" error="Invalid account number" touched={false}>
        <input />
      </FormField>,
    );

    expect(queryByText(/Invalid account number/)).toBeNull();
  });

  it('matches snapshot', () => {
    const tree = renderer
      .create(
        <FormField label="Account Number" error={undefined} touched={false}>
          <input />
        </FormField>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
