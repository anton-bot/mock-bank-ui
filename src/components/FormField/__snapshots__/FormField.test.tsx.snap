// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Field matches snapshot 1`] = `
<div
  className="FormField"
>
  <div
    className="label"
  >
    Account Number
  </div>
  <div
    className="control"
  >
    <input />
  </div>
</div>
`;
